parla/device.py

```diff
@@ -109,8 +109,8 @@ def __init__(self, architecture: "Architecture", index, *args, **kwds):
         """
         Construct a new Device with a specific architecture.
         """
-        self.architecture = architecture
-        self.index = index
+        self.architecture = architecture  # typically parla.cpu_impl.cpu or parla.cuda.gpu
```

Comment:
Please remove typically.
Instead add TODO().
e.g. TODO(yineng): more architectures could be added in the future